Mathlib/Topology/ContinuousMap/Interval.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2024 Vincent Beffara. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vincent Beffara
+-/
+import Mathlib.Topology.CompactOpen
+import Mathlib.Topology.Order.ProjIcc
+
+/-!
+# Continuous bundled maps on intervals
+
+In this file we prove a few results about `ContinuousMap` when the domain is an interval.
+-/
+
+open Set ContinuousMap Filter Topology
+
+namespace ContinuousMap
+
+variable {α : Type*} [LinearOrder α] [TopologicalSpace α] [OrderTopology α] {a b c : α}
+variable {E F : Type*} [TopologicalSpace E] [TopologicalSpace F]
+
+/-- The embedding into an interval from a sub-interval lying on the left, as a `ContinuousMap`. -/
+def subinterval_left (h : b ∈ Icc a c) : C(Icc a b, Icc a c) where
+  toFun x := ⟨x, x.2.1, x.2.2.trans h.2⟩
+
+/-- The embedding into an interval from a sub-interval lying on the right, as a `ContinuousMap`. -/
+def subinterval_right (h : b ∈ Icc a c) : C(Icc b c, Icc a c) where
+  toFun x := ⟨x, h.1.trans x.2.1, x.2.2⟩
+
+/-- The value of a `ContinuousMap` defined on an interval at the left extremity of the interval. -/
+def leftval (hab : a ≤ b) (f : C(Icc a b, E)) : E :=
+  f ⟨a, le_rfl, hab⟩
+
+/-- The value of a `ContinuousMap` defined on an interval at the right extremity of the interval. -/
+def rightval (hbc : b ≤ c) (f : C(Icc b c, E)) : E :=
+  f ⟨c, hbc, le_rfl⟩
+
+/-- The map `leftval` as a `ContinuousMap`. -/
+def leftvalCM (hab : a ≤ b) : C(C(Icc a b, E), E) :=
+  ⟨leftval hab, continuous_eval_const _⟩
+
+/-- The map `rightval` as a `ContinuousMap`. -/
+def rightvalCM (hbc : b ≤ c) : C(C(Icc b c, E), E) :=
+  ⟨rightval hbc, continuous_eval_const _⟩
```

Comment:
And then these become the evaluation map at `⊥` and `⊤`, so probably also not worth having as separate definitions

Reply:
Right.